### PR TITLE
Explosion sounds

### PR DESCRIPTION
**Pull request by Anewbe**
Should fix them.

### CODE REVIEW

**Comment by SilverTalismen** [comment]
Betcha had a real *Blast* trying to fix this huh? c:

File: code/game/objects/explosion.dm [hunk 1]
```diff
@@ -44,6 +44,13 @@ proc/explosion(turf/epicenter, devastation_range, heavy_impact_range, light_impa
 					far_volume += (dist <= far_dist * 0.5 ? 50 : 0) // add 50 volume if the mob is pretty close to the explosion
 					M.playsound_local(epicenter, 'sound/effects/explosionfar.ogg', far_volume, 1, frequency, falloff = 5)
 
+		var/close = range(world.view+round(devastation_range,1), epicenter)
+		// to all distanced mobs play a different sound
+		for(var/mob/M in world) if(M.z == epicenter.z) if(!(M in close))
+			// check if the mob can hear
+			if(M.ear_deaf <= 0 || !M.ear_deaf) if(!istype(M.loc,/turf/space))
```
Comment (RemieRichards) [inline code/game/objects/explosion.dm]: I am quite amused that this line compiled.

Comment (Anewbe) [inline code/game/objects/explosion.dm]: Oh hey, weird.

File: code/game/objects/explosion.dm [hunk 2]
```diff
@@ -44,6 +44,14 @@ proc/explosion(turf/epicenter, devastation_range, heavy_impact_range, light_impa
 					far_volume += (dist <= far_dist * 0.5 ? 50 : 0) // add 50 volume if the mob is pretty close to the explosion
 					M.playsound_local(epicenter, 'sound/effects/explosionfar.ogg', far_volume, 1, frequency, falloff = 5)
 
+		var/close = range(world.view+round(devastation_range,1), epicenter)
+		// to all distanced mobs play a different sound
+		for(var/mob/M in world) if(M.z == epicenter.z) if(!(M in close))
+			// check if the mob can hear
+			if(M.ear_deaf <= 0 || !M.ear_deaf)
```
Comment (RemieRichards) [inline code/game/objects/explosion.dm]: While the last time was complete nonsense code that didn't deserve to compile, you CAN do this in a single line+if.
```if((M.ear_deaf <= 0 || !M.ear_deaf)) && !istype(M.loc, /turf/space))``
Though, you most likely want to swap your current ```M.loc``` for ```get_turf(M)```

As to WHY the last try compiled, I'd assume (token wise) it became:
```
if(M.ear_deaf <= 0 || !M.ear_deaf)
    if(!istype(M.loc, /turf/space))
        ...
```
Which, amusingly, is what you have written here.

**Comment by Anewbe** [comment]
Being told to merge this.

**Pull request merged**